examples/vision/depth_estimation.py

```diff
@@ -0,0 +1,507 @@
+"""
+
+
+class DataGenerator(tf.keras.utils.Sequence):
+    def __init__(self, data, batch_size=6, dim=(768, 1024), n_channels=3, shuffle=True):
+        """
+        Initialization
+        """
+        self.data = data
+        self.indices = self.data.index.tolist()
+        self.dim = dim
+        self.n_channels = n_channels
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.on_epoch_end()
+
+    def __len__(self):
+        return int(np.ceil(len(self.data) / self.batch_size))
+
+    def __getitem__(self, index):
+        if (index + 1) * self.batch_size > len(self.indices):
+            self.batch_size = len(self.indices) - index * self.batch_size
+        # Generate one batch of data
+        # Generate indices of the batch
+        index = self.indices[index * self.batch_size : (index + 1) * self.batch_size]
+        # Find list of IDs
+        batch = [self.indices[k] for k in index]
+        X, y = self.__data_generation(batch)
+
+        return X, y
+
+    def on_epoch_end(self):
+
+        """
+        Updates indexes after each epoch
+        """
+        self.index = np.arange(len(self.indices))
+        if self.shuffle == True:
+            np.random.shuffle(self.index)
+
+    def __load__(self, image_path, depth_map, mask):
+        "Load Target image"
+
+        image_ = cv2.imread(image_path)
+        image_ = cv2.cvtColor(image_, cv2.COLOR_BGR2RGB)
+        image_ = cv2.resize(image_, self.dim)
+        image_ = (image_) / 255.0
+
+        depth_map = np.load(depth_map).squeeze()
+
+        mask = np.load(mask)
+        mask = mask > 0
+
+        MIN_DEPTH = 0.1
+
+        MAX_DEPTH = min(300, np.percentile(depth_map, 99))
+        depth_map = np.clip(depth_map, MIN_DEPTH, MAX_DEPTH)
+        depth_map = np.log(depth_map, where=mask)
+
+        depth_map = np.ma.masked_where(~mask, depth_map)
+
+        depth_map = np.clip(depth_map, 0.1, np.log(MAX_DEPTH))
+        depth_map = cv2.resize(depth_map, self.dim)
+        depth_map = np.expand_dims(depth_map, axis=2)
+
+        # depth_map = tf.image.convert_image_dtype(depth_map, tf.float32)
+
+        return image_, depth_map
+
+    def __data_generation(self, batch):
+
+        X = np.empty((self.batch_size, *self.dim, self.n_channels))
+        y = np.empty((self.batch_size, *self.dim, 1))
+
+        for i, id_ in enumerate(batch):
+            X[i,], y[i,] = self.__load__(
+                self.data["image"][id_],
+                self.data["depth"][id_],
+                self.data["mask"][id_],
+            )
+
+        return X, y
+
+
+"""
+## Visualizing Samples
+"""
+
+visualize_samples = next(
+    iter(DataGenerator(data=df, batch_size=6, dim=(config.HEIGHT, config.WIDTH)))
+)
+input, target = visualize_samples
+cmap = plt.cm.jet
+cmap.set_bad(color="black")
+fig, ax = plt.subplots(6, 2, figsize=(50, 50))
+for i in range(6):
+    ax[i, 0].imshow((input[i].squeeze()))
+    ax[i, 1].imshow((target[i].squeeze()), cmap=cmap)
```

Comment:
Please update this code, it gonna make unpleasant display.
More-ever you are using it three times so it would be better if you make a reusable function for it.

Reply:
updated